README.md

```diff
@@ -1,2 +1,323 @@
+
+## Installation
+
+Edx-Adapt application requires
+[MongoDB](https://docs.mongodb.com/manual/administration/install-on-linux/)
+database.
+
+Clone edx-adapt project to your server and switch to the project
+directory:
+
+```bash
+> git clone git@github.com:raccoongang/edx-adapt.git@<tag_stable_version>
+> cd edx-adapt
+```
+
+Create virtual environment:
+
+```bash
+> virtualenv env_name
+```
+
+Activate virtual environment and install all requirements:
+
+```bash
+> source env_name/bin/activate
+> pip install -r requirements
+```
+
+## Run edx-adapt application in development mode:
+
+```
+> python edx_adapt.py
+```
+
+
+## Run edx-adapt application in production mode
+
+There are few sample config files in `etc/folder` required to configure
+production environment.
+
+`etc/nginx/` contains template file to configure nginx server
+
+`etc/edx_adapt/` contains template file to configure uwsgi server for
+edx-adapt
+
+`etc/init/` contains template file to configure edx-adapt be proceed by
+service manager
+
+## Main API endpoints
+
+`/api/v1/course`
+
+- GET: Show all courses registered in Edx-Adapt
+ - `response.data = {'course_ids': courses}`
+- POST: Create new course in Edx-Adapt
+ - Parameters: `course_id` (string)
+
+
+`/api/v1/course/<course_id>/skill`
+
+- GET: Show all skills which course `<course_id>` contains.
+ - `response.data = {'skills': skills}`
+- POST: Add skill into course `<course_id>`
+ - Parameters : `skill_name` (string)
+
+
+`/api/v1/course/<course_id>/user`
+
+- GET: Show all users registered in the course `<course_id>`
+ - `response.data = {'users': {'finished': finished_users, 'in_progress':
+progress_users}}`
+- POST: Enroll new user in the course `<course_id>`
+ - Parameters: `user_id` (string)
+
+
+`/api/v1/course/<course_id>`
+
+- GET: Show all problems contained to the course `<course_id>`
+ - `response.data = {'problems': problems}`
+- POST: Add new problem in the course `<course_id>`
+ - Parameters: `problem_name` (string), `tutor_url` (string), `skills`
+(list of strings), `pretest` (boolean), `posttest` (boolean)
+
+
+`/api/v1/course/<course_id>/skill/<skill_name>`
+
+- GET: Show all problems contained in the course `<course_id>` related
+to the skill `<skill_name>`
+ - `response.data = {'problems': problems}`
+
+
+`/api/v1/course/<course_id>/experiment`
+
+- GET: Show result of all experiments for the course `<course_id>`
+ - `response.data = {'experiments': exps}`
+
+
+`/api/v1/course/<course_id>/probabilities`
+
+- GET: Show list with default BKT model parameters added to the course
+`<course_id>`
+ - `response.data = {'model_params': prob_list}`
+
+- POST: Add new or update existing model parameters in the course
+`<course_id>`
+ - Parameters: `prob_list` (list of dicts with models parameters)
+`[{threshold: float, pg: float, ps: float, pi: float, pt: float}, ...]`
+
+
+`/api/v1/course/<course_id>/user/<user_id>/interaction`
+
+- POST: Add user’s interaction with Edx `->` Edx-Adapt
```

Comment:
Oh, sorry for confusion, I was talking about apostrophes/single quotes.
`’` -> `'`